account/lnurl.go

```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
 	Token string `json:"token"`
 }
 
-func (a *Service) HandleLNURL(rawString string) (*data.LNUrlResponse, error) {
+func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err error) {
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
+	if ok {
+		a.log.Infof("HandleLNURL %v", encodedLnurl)
+		rawurl, iparams, err = lnurl.HandleLNURL(encodedLnurl)
```

Comment:
In general in go it is a convention to return error immediately when we have one as it makes a more clear flow.
I think here we should check for error and return error immediately